src/screens/SendScreen.tsx

```diff
@@ -0,0 +1,361 @@
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import {
+  APIError,
+  convertAlphToSet,
+  formatAmountForDisplay,
+  isAddressValid,
+  MINIMAL_GAS_AMOUNT,
+  MINIMAL_GAS_PRICE
+} from '@alephium/sdk'
+import { SweepAddressTransaction } from '@alephium/sdk/api/alephium'
+import { StackScreenProps } from '@react-navigation/stack'
+import { Codesandbox } from 'lucide-react-native'
+import { useCallback, useEffect, useState } from 'react'
+import { Alert, ScrollView, View } from 'react-native'
+import Toast from 'react-native-root-toast'
+import styled, { useTheme } from 'styled-components/native'
+
+import client from '../api/client'
+import Amount from '../components/Amount'
+import AppText from '../components/AppText'
+import Button from '../components/buttons/Button'
+import ButtonsRow from '../components/buttons/ButtonsRow'
+import ExpandableRow from '../components/ExpandableRow'
+import HighlightRow from '../components/HighlightRow'
+import InfoBox from '../components/InfoBox'
+import AddressSelector from '../components/inputs/AddressSelector'
+import Input from '../components/inputs/Input'
+import Screen, {
+  BottomModalScreenTitle,
+  BottomScreenSection,
+  ScreenSection,
+  ScreenSectionTitle
+} from '../components/layout/Screen'
+import ModalWithBackdrop from '../components/ModalWithBackdrop'
+import { useAppDispatch, useAppSelector } from '../hooks/redux'
+import InWalletTabsParamList from '../navigation/inWalletRoutes'
+import RootStackParamList from '../navigation/rootStackRoutes'
+import { addPendingTransactionToAddress, selectAddressByHash } from '../store/addressesSlice'
+import { AddressHash } from '../types/addresses'
+import { isNumericStringValid } from '../utils/numbers'
+
+type ScreenProps = StackScreenProps<InWalletTabsParamList & RootStackParamList, 'SendScreen'>
+
+type TxStep = 'build' | 'send'
+
+const SendScreen = ({
+  navigation,
+  route: {
+    params: { addressHash }
+  }
+}: ScreenProps) => {
+  const theme = useTheme()
+  const mainAddress = useAppSelector((state) => state.addresses.mainAddress)
+  const [fromAddressHash, setFromAddressHash] = useState<AddressHash>(addressHash ?? mainAddress)
+  const fromAddress = useAppSelector((state) => selectAddressByHash(state, fromAddressHash))
+  const [toAddressHash, setToAddressHash] = useState<string>('')
+  const [amountString, setAmountString] = useState('')
+  const [amount, setAmount] = useState(BigInt(0))
+  const [fees, setFees] = useState<bigint>(BigInt(0))
+  const [unsignedTxId, setUnsignedTxId] = useState('')
+  const [unsignedTransaction, setUnsignedTransaction] = useState('')
+  const [gasAmount, setGasAmount] = useState('')
+  const [gasPriceString, setGasPriceString] = useState('')
+  const [gasPrice, setGasPrice] = useState<bigint>()
+  const [isLoading, setIsLoading] = useState(false)
+  const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
```

Comment:
That's a lot of states 🙃
I wonder if we should maybe start considering using something like https://react-hook-form.com/ which would greatly simplify local form state management and validation.

Reply:
I'll have a look into it!

Reply:
Hm, looking at the [integration process of React Hook Form into React Native](https://echobind.com/post/react-hook-form-for-react-native) and the changes that are needed to our codebase, I am not so sure it's worth going for it for one component that has a lot of state. WDYT? I could try reducing the use of `useState` by grouping things together...

Reply:
Or maybe I am wrong, wait: https://react-hook-form.com/get-started#reactnative